app/lib/frontend/backend.dart

```diff
@@ -743,10 +743,10 @@ class GCloudPackageRepository extends PackageRepository {
   }
 
   void _validatePackageUploader(
-      models.Package package, String userId, String userEmail) {
+      String packageName, models.Package package, String userId) {
     // Fail if package doesn't exist.
     if (package == null) {
-      throw GenericProcessingException('Package "$package" does not exist');
+      throw NotFoundException.resource(packageName);
```

Comment:
Don't we need to make sure that `NotFoundException` implements `GenericProcessingException` such that `package:pub_server` does the right thing? Or is this a custom error we define in pub-dev?

Reply:
Good question. `pub_server` will return a 400 with a slightly different JSON structure:
`{'error': {'message': message}}`.

I'd do the `implements GenericProcessingException` for now, and we should consider if/how we want to standardize the message formats.